osquery/examples/example_writable_table.cpp

```diff
@@ -0,0 +1,222 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/sdk.h>
+#include <osquery/system.h>
+
+#include <rapidjson/document.h>
+
+#include <sstream>
+
+using namespace osquery;
+class WritableTable : public TablePlugin {
+ private:
+  /// Contents for a single row
+  struct RowData final {
```

Comment:
Can you elaborate on why you chose to use a `struct` here instead of an `unordered_map`? It seems to me that if these were maps you could just return them in your gen function instead of copying the values to the `Row` object. Maybe you can just use `Row` objects as well? Another upside is that most of your int conversions will go away.

The only draw back I see is not being able to store types in the their native types (they all become strings).

Let me know if I missed something obvious about your implementation :p

Reply:
I couldn't use strings because the data is now serialized using rapidjson, automatically matching the type of each column